tests/test_downloadermiddleware_httpcompression.py

```diff
@@ -144,3 +144,13 @@ def test_process_response_gzipped_contenttype(self):
         self.assertIs(newresponse, response)
         self.assertEqual(response.headers['Content-Encoding'], b'gzip')
         self.assertEqual(response.headers['Content-Type'], b'application/gzip')
+
+    def test_process_response_head_request_no_decode_required(self):
+        response = self._getresponse('gzip')
+        response.headers['Content-Type'] = 'application/gzip'
+        request = response.request
+        request.method = 'HEAD'
+        response = response.replace(body = None)
+        newresponse = self.mw.process_response(request, response, self.spider)
+        self.assertIs(newresponse, response)
+        self.assertEquals(response.body, '')
```

Comment:
The tests are failing in Python 3 because `''` is not bytes, but a native string (which in Python 3 is the same as unicode object in Python 2).
If you update this to `b''` tests should pass.